packetbeat/flows/worker.go

```diff
@@ -136,11 +151,50 @@ func newFlowsWorker(pub Reporter, watcher *procs.ProcessesWatcher, table *flowMe
 		return nil, ErrInvalidPeriod
 	}
 
+	tick, ticksTimeout, ticksPeriod, ticksActiveTimeout := getTicksAndTimeouts(timeout, period, activeTimeout)
+
+	debugf("new flows worker. timeout=%v, period=%v, tick=%v, ticksTO=%v, ticksP=%v, activeTimeout=%v, ticksActiveTO=%v",
+		timeout, period, tick, ticksTimeout, ticksPeriod, activeTimeout, ticksActiveTimeout)
+
+	defaultBatchSize := 1024
+	processor := &flowsProcessor{
+		table:    table,
+		watcher:  watcher,
+		counters: counters,
+		timeout:  timeout,
+	}
+	processor.spool.init(pub, defaultBatchSize)
+
+	return makeWorker(processor, tick, ticksTimeout, ticksPeriod, ticksActiveTimeout, 10)
+}
+
+func getTicksAndTimeouts(timeout, period, activeTimeout time.Duration) (time.Duration, int, int, int) {
 	tick := timeout
 	ticksTimeout := 1
+	ticksActiveTimeout := -1
 	ticksPeriod := -1
+
+	// If activeTimeout is set, we need to calculate the tick for the worker
+	// TODO: I think these two if conditions can maybe be represented in a better way
+	if activeTimeout > 0 {
```

Comment:
Im having trouble following this logic - is it possible to add a comment describing what is happening in this block?